### PR TITLE
inventory/azure_rm.py: Allow Unix environment variables to override .ini file settings

### PR DESCRIPTION
**Pull request by scottkof**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
contrib/inventory/azure_rm.py

##### ANSIBLE VERSION
```
Traceback (most recent call last):
  File "bin/ansible", line 43, in <module>
    import ansible.constants as C
ImportError: No module named ansible.constants
```

##### SUMMARY
Azure dynamic inventory only reads settings from Unix environment variables if the .ini settings file doesn't exist. This PR modifies the script so both file settings and env vars are always read, with env vars taking precedence. This PR fixes the script's surprising, non-standard behavior.

### CODE REVIEW

**Comment by gundalow** [comment]
@scottkof Thanks for the PR, your change sounds sensible. I'll hand over  to @nitzmahone and @chouseknecht who have made changes to this file before for review & merge

**Comment by yuwzho** [comment]
As previous comment, waiting @nitzmahone and @chouseknecht . As credential configuration priority chain issue, what's the standard way ansible inventory does for the option configuration chain?

**Comment by Fred-sun** [comment]
HI @chouseknecht @nitzmahone , This PR has not been updated for more than twenty months, It need you review and approve, If you have other view, Please update your point and next step plan? Thanks!

**Comment by Fred-sun** [comment]
@chouseknecht @nitzmahone , Could you help review this PR? Thanks!

**Comment by chouseknecht** [comment]
The code looks sensible to me.

Shipit

**Comment by Fred-sun** [comment]
@chouseknecht @nitzmahone Thanks for you contribution, @nitzmahone Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone This PR has open more than one year, Could you help review this PR and push to next step forward? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone This PR has open more than one year, Could you help review this PR and push to next step forward? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone This PR has open more than one year, Could you help review this PR and push to next step forward? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR and push to forward merge? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR and push to forward merge? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR and push to forward merge? too old. Thanks!

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
Updating existing azure_rm.py inventory to leverage cloud auth should be in seperate PR. This PR is fixing bug that with .ini file, environment variables not working. Fixing looks good

**Comment by yungezz** [comment]
shipit

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR and push to forward? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone

**Comment by jborean93** [comment]
All PRs targeted towards the older Azure inventory script need to be looked at and need to be closed if we wish to continue development on it. In Ansible 2.7 a new inventory plugin for Azure is the way forward and is designed to overcome all the shortfalls in the current script. You can find info on this new plugin here https://docs.ansible.com/ansible/devel/plugins/inventory/azure_rm.html

**Comment by yungezz** [comment]
@jborean93 got it. close PRs on old inventory means closing them without merging? suggest existing user move to new inventory? i'm little concerned aboiut existing user who used old inventory already. Is there any announcement or deprecated plan so people could be aware of the new inventory?

**Comment by jborean93** [comment]
@yungezz that’s what needs to be discussed. I think we may want to look at bug fixes but there is no use in adding new features to something we wish to get rid off.

**Comment by yungezz** [comment]
@jborean93  sure, let's discuss it in next syncup meeting

**Comment by Fred-sun** [comment]
@yungezz @jborean93 Could you help confirm how the PR is going? further review or merged? Thanks!

**Comment by Fred-sun** [comment]
@yungezz @jborean93

**Comment by Fred-sun** [comment]
@yungezz @jborean93 Could you help recheck the PR? How about the PR going ahead? need merged or further review. Thanks!

**Comment by jborean93** [comment]
Ok I'm sorry for having this PR languish over time, unfortunately we are not going to go ahead with this one. As per https://github.com/ansible/ansible/pull/46939 this script has been superseeded by the new Azure inventory plugin. Documentation and info for this plugin can be found at https://docs.ansible.com/ansible/latest/plugins/inventory/azure_rm.html and it was added in the 2.7 release.

There is nothing stopping people from using their own contrib script but Ansible will only be focusing on the inventory plugin for all feature requests. Only pull requests that contain major bug fixes for `contrib/inventory/azure_rm.py` will be reviewed and accepted.

**Pull request closed without merging**
